File: R/80-remove-duplicates-from-sorted-array-ii/main.go
package main

import (
	"fmt"
)

func removeDuplicates(nums []int) int {
	if len(nums) <= 2 {
		return len(nums)
	}
	offset := 0
	c := 1
	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[offset] {
			offset++
			nums[offset] = nums[i]
			c = 1
		} else if c == 1 {
			c++
			offset++
			nums[offset] = nums[i]
		}
	}

	return offset + 1
}

func main() {
	fmt.Println(removeDuplicates(([]int{1, 1, 1, 2, 2, 3})))
}

File: R/80-remove-duplicates-from-sorted-array-ii/main.rs
struct Solution {}

impl Solution {
  pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
    if nums.len() <= 2 {
      return nums.len() as _;
    }
    let mut start: usize = 2;
    let mut end: usize = 2;
    (2..nums.len()).for_each(|idx| {
      if nums[start - 2] != nums[idx] {
        nums[start] = nums[idx];
        start += 1;
      }
    });
    start as _
  }
}
